src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -98,19 +100,22 @@ def run(
         :return: The result of this batch run
         :rtype: ~promptflow.batch._result.BatchResult
         """
-        self._start_time = datetime.utcnow()
-        # set batch input source from input mapping
-        OperationContext.get_instance().set_batch_input_source_from_inputs_mapping(inputs_mapping)
-        # resolve input data from input dirs and apply inputs mapping
-        batch_input_processor = BatchInputsProcessor(self._working_dir, self._flow.inputs, max_lines_count)
-        batch_inputs = batch_input_processor.process_batch_inputs(input_dirs, inputs_mapping)
-        # run flow in batch mode
-        output_dir = resolve_dir_to_absolute(self._working_dir, output_dir)
-        with _change_working_dir(self._working_dir):
-            batch_result = self._exec_batch(batch_inputs, run_id, output_dir, raise_on_line_failure)
-        # destroy executor proxy
-        self._executor_proxy.destroy()
-        return batch_result
+
+        try:
+            self._start_time = datetime.utcnow()
+            # set batch input source from input mapping
+            OperationContext.get_instance().set_batch_input_source_from_inputs_mapping(inputs_mapping)
+            # resolve input data from input dirs and apply inputs mapping
+            batch_input_processor = BatchInputsProcessor(self._working_dir, self._flow.inputs, max_lines_count)
+            batch_inputs = batch_input_processor.process_batch_inputs(input_dirs, inputs_mapping)
+            # run flow in batch mode
+            output_dir = resolve_dir_to_absolute(self._working_dir, output_dir)
+            with _change_working_dir(self._working_dir):
+                batch_result = self._exec_batch(batch_inputs, run_id, output_dir, raise_on_line_failure)
+            return batch_result
+        finally:
+            # destroy executor proxy
```

Comment:
Better to catch and log exceptioni here? Then raise.

Reply:
Updated.